src/dodal/devices/i18/table.py

```diff
@@ -0,0 +1,28 @@
+from ophyd_async.core import (
+    AsyncStatus,
+    StandardReadable,
+)
+from ophyd_async.epics.motor import Motor
+from pydantic import BaseModel
+
+
+class Four_D_Position(BaseModel):
+    x: float
+    y: float
+    z: float
+    theta: float
```

Comment:
```suggestion
class TablePosition(BaseModel):
    x: float
    y: float
    z: float | None = None
    theta: float | None = None
```

https://peps.python.org/pep-0008/#class-names

And allows for the case where your `set` method only cares about x, y